### PR TITLE
feat: there are some bug

### PR DESCRIPTION
**Pull request by GuhaPritam**

### CODE REVIEW

File: student-marks-board/index.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
+let userData = [];
+
+function fetchData() {
+	fetch('./data/users.json')
+		.then((response) => response.json())
+		.then((data) => {
+			const users = data.users;
+			userData = users;
+			createUI(users[0]['standard_1']);
```
Comment (atapas) [inline student-marks-board/index.js]: Instead of passing `users[0]['standard_1']`, `pass only users[0]`

File: student-marks-board/index.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+		});
+}
+fetchData();
+
+function createUI(users) {
+	const mainDiv = document.getElementById('tableDiv');
```
Comment (atapas) [inline student-marks-board/index.js]:
```js
let obj = {
  'standard_1': [],
  'standard_2': [],
  'standard_3': [],
....
....
}
```

Do 
```js
let keys = obj.keys(); // [standard_1, standard_2, standard_3]

// loop the above array

```

obj[keys[0]]

**Review by atapas — CHANGES_REQUESTED** [review]

File: student-marks-board/index.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+	const ul = document.createElement('ul');
+	mainDiv.appendChild(ul);
+
+    let keys = userData.keys('userData');
+
+    for(let j = 0; j <= keys - 1; j++) {
+        for (let i = 0; i <= users.length - 1; i++) {
+            const li = document.createElement('li');
+            li.setAttribute('class','userLi');
+            li.innerHTML = `<span>${users[j],[i].name}</span><span>${users[i].grade}</span>`;
```
Comment (atapas) [inline student-marks-board/index.js]: This is wrong `${users[j],[i].name}` what is this doing?

**Review by atapas — CHANGES_REQUESTED** [review]

**Comment by GuhaPritam** [comment]
ami purotai change kore push korechi

File: student-marks-board/index.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+	const ul = document.createElement('ul');
+	mainDiv.appendChild(ul);
+
+    let keys = userData.keys('userData');
+
+    for(let j = 0; j <= keys - 1; j++) {
+        for (let i = 0; i <= users.length - 1; i++) {
+            const li = document.createElement('li');
+            li.setAttribute('class','userLi');
+            li.innerHTML = `<span>${users[i].name}</span><span>${users[i].grade}</span>`;
```
Comment (atapas) [inline student-marks-board/index.js]: Wrong again.. Does your `users[i]` has the `name` property? console log users[i] and debug and check..

Hint: You are not using keys.

**Review by atapas — CHANGES_REQUESTED** [review]

File: student-marks-board/index.js [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+	mainDiv.appendChild(ul);
+
+	for (let i = 0; i <= users.length - 1; i++) {
+		const li = document.createElement('li');
+		li.setAttribute('class', 'userLi');
+		li.innerHTML = `<span>${users[i].name}</span><span>${users[i].grade}</span>`;
+		ul.appendChild(li);
+	}
+}
+
```
Comment (atapas) [inline student-marks-board/index.js:33]: what do you want me to review?

**Review by atapas — CHANGES_REQUESTED** [review]

File: student-marks-board/index.js [hunk 6]
```diff
@@ -0,0 +1,51 @@
+let userData = [];
+
+function fetchData() { 
+	fetch('./data/users.json') //this is data of json
+		.then((response) => response.json()) //get successfully data then i got a response and i take json data
+		.then((data) => { //this is my all data
+			const users = data.users;//data is an array and users is a variable where i store my data
+			userData = users;//save users input in js object
+
+			const keys = Object.keys(users);//get the value from users
```
Comment (atapas) [inline student-marks-board/index.js:10]: I am not clear what is `keys`. Please explain.

File: student-marks-board/index.js [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+	for (let i = 0; i <= users.length - 1; i++) {
+		const li = document.createElement('li');
+		li.setAttribute('class', 'userLi');
+		li.innerHTML = `<span>${users[i].name}</span><span>${users[i].grade}</span>`;
+		ul.appendChild(li);//put ul in li
+	}
+}
+
+function searchFun(event) {
+	const term = event.target.value;
```
Comment (atapas) [inline student-marks-board/index.js]: Add comments.

**Review by atapas — CHANGES_REQUESTED** [review]

**Pull request closed without merging**